### PR TITLE
iOS-1811 Update subscription service

### PR DESCRIPTION
**Pull request by mgolovko**

### CODE REVIEW

File: Anytype/Sources/Models/Documents/Events/Subscriptions/Internals/SubscriptionStorage.swift [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+            guard events.contextId.isEmpty else { return }
+            await self?.handle(events: events)
+        }
+    }
+    
+    deinit {
+        guard let data else { return }
+        Task { [data, toggler] in
+            try await toggler.stopSubscription(id: data.identifier)
+        }
```
Comment (joe-pusya) [inline Anytype/Sources/Models/Documents/Events/Subscriptions/Internals/SubscriptionStorage.swift:46]: you can call `stopSubscription()` here

File: Anytype/Sources/PresentationLayer/Modules/HomeWidgets/Widgets/List/Specific/List/SetListWidgetModuleAssembly.swift [hunk 2]
```diff
@@ -27,7 +27,7 @@ final class SetListWidgetModuleAssembly: HomeWidgetCommonAssemblyProtocol {
             setSubscriptionDataBuilder: SetSubscriptionDataBuilder(
                 activeWorkspaceStorage: serviceLocator.activeWorkspaceStorage()
             ),
-            subscriptionService: serviceLocator.subscriptionService(),
+            subscriptionStorageProvider: serviceLocator.subscriptionStorageProvider(),
```
Comment (joe-pusya) [inline Anytype/Sources/PresentationLayer/Modules/HomeWidgets/Widgets/List/Specific/List/SetListWidgetModuleAssembly.swift:30]: `reminder`: use subscription service in my current PR with templates - needs to be updated too (by me or you)

File: Anytype/Sources/PresentationLayer/TextEditor/Set/EditorSetViewModel.swift [hunk 3]
```diff
@@ -851,7 +864,8 @@ extension EditorSetViewModel {
             ),
             interactor: DI.preview.serviceLocator.objectHeaderInteractor(objectId: "objectId")
         ),
-        subscriptionService: DI.preview.serviceLocator.subscriptionService(),
+//        subscriptionService: DI.preview.serviceLocator.subscriptionService(),
```
Comment (joe-pusya) [inline Anytype/Sources/PresentationLayer/TextEditor/Set/EditorSetViewModel.swift:867]: remove?

**Review by joe-pusya — APPROVED** [review]
❤️‍🔥

File: Anytype/Sources/Models/Documents/Events/Subscriptions/Internals/SubscriptionStorage.swift (hunk 1 above)
Comment (mgolovko) [reply Anytype/Sources/Models/Documents/Events/Subscriptions/Internals/SubscriptionStorage.swift:46]: This is deinit. And if i call `stopSubscription` there will be a crash

File: Anytype/Sources/PresentationLayer/TextEditor/Set/EditorSetViewModel.swift (hunk 3 above)
Comment (mgolovko) [reply Anytype/Sources/PresentationLayer/TextEditor/Set/EditorSetViewModel.swift:867]: yes!

File: Anytype/Sources/Models/Documents/Events/Subscriptions/Internals/SubscriptionStorage.swift (hunk 1 above)
Comment (mgolovko) [reply Anytype/Sources/Models/Documents/Events/Subscriptions/Internals/SubscriptionStorage.swift:46]: self shouldn't be capture

Comment (joe-pusya) [reply Anytype/Sources/Models/Documents/Events/Subscriptions/Internals/SubscriptionStorage.swift:46]: didn’t notice, agree!

**Pull request merged**